src/modules/Game.class.js

```diff
@@ -20,25 +20,297 @@ class Game {
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.initialState = initialState;
+    this.state = JSON.parse(JSON.stringify(initialState));
+    this.status = 'idle';
+    this.score = 0;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  moveLeft() {
+    const state = this.getState();
+
+    // below I update the state if ArrowLeft was pressed
+
+    for (let row = 0; row < 4; row++) {
+      const cache = [];
+
+      // here I push all digits (except 0) of current row in the state
+      // to the cache array
+
+      for (let cell = 0; cell < 4; cell++) {
+        const currentDigit = state[row][cell];
+
+        if (!currentDigit) {
+          continue;
+        }
+
+        cache.push(currentDigit);
+      }
+
+      // here I sum equal numbers in the cache array and update the score
+
+      for (let i = 0; i < cache.length; i++) {
+        const currentDigit = cache[i];
+        const nextDigit = i + 1 < cache.length ? cache[i + 1] : null;
+
+        if (!nextDigit) {
+          break;
+        }
+
+        if (currentDigit === nextDigit) {
+          const prevScore = this.getScore();
+          const currentScore = prevScore + currentDigit * 2;
+
+          this.score = currentScore;
+
+          cache.splice(i, 2, currentDigit * 2);
+        }
+      }
+
+      // if row doesn't change then go to the next row
+
+      if (cache.length === 4) {
+        continue;
+      }
+
+      // here I need to fill cache array with 0 for equality cache.length === 4
+
+      while (cache.length < 4) {
+        cache.push(0);
+      }
+
+      // here I rewrite current row in the state
+
+      for (let cell = 0; cell < 4; cell++) {
+        state[row][cell] = cache[cell];
+      }
+    }
+  }
+
+  moveRight() {
+    const state = this.getState();
+
+    // below I update the state if ArrowRight was pressed
+
+    for (let row = 0; row < 4; row++) {
+      const cache = [];
+
+      // here I push all digits (except 0) of current row in the state
+      // to the cache array
+
+      for (let cell = 0; cell < 4; cell++) {
+        const currentDigit = state[row][cell];
+
+        if (!currentDigit) {
+          continue;
+        }
+
+        cache.push(currentDigit);
+      }
+
+      // here I sum equal numbers in the cache array and update the score
+
+      // if we press ArrowRight, digits should sum from right/end to left/start
+      // [4, 4, 4] -> [4, 8]
```

Comment:
you can simplify moveRight, just rotate the state, call the moveLeft method, and then rotate the state again, it should work like moveRight